src/model/SourceOperation.js

```diff
@@ -0,0 +1,58 @@
+import Ressource from "./Ressource";
+import Result from "./Result";
+import { getConfig } from "../config";
+import _urlParser from "../urlParser";
+import request from "../api";
+
+const paramDefaults = {};
+const _queryUrl =
+  "/projects/:projectId/environments/:environmentId/source-operations";
+const _url =
+  "/projects/:projectId/environments/:environmentId/source-operation";
+
+export default class SourceOperation extends Ressource {
+  constructor(sourceOperation, url) {
+    super(url, paramDefaults, {}, sourceOperation);
+    this.operation = "";
+    this.app = "";
+    this.commmand = "";
+  }
+
+  // This is a custom method because we have to override the url
+  // that gets passed into the prototype.constructor().
+  static query(params) {
+    const { api_url } = getConfig();
+    const queryUrl = _urlParser(
+      `${api_url}${_queryUrl}`,
+      params,
+      paramDefaults
+    );
+
+    const url = _urlParser(`${api_url}${_url}`, params, paramDefaults);
```

Comment:
You could just use the queryUrl in the constructor as well right?

Reply:
I tried that and because of the way builds the URL it won't work for this. It wants to append an `/:id` to the end of the URL. For this model we only a single `/source-operations` (no `id`) endpoint, and an endpoint to trigger a SourceOp with `POST /source-operation` (note no `s`). The way `._url` and `._queryUrl` are built they don't allow us to achieve the correct URL format.